scripts/launch-s0-instance.sh

```diff
@@ -0,0 +1,95 @@
+security_group="elastio-s0-server"
+default_instance_type="m5d.2xlarge"
+default_instance_name="$(whoami)-s0-server"
+
+usage()
+{
+    echo "Usage examples:"
+    echo "   $me --key-name my_ec2_key --name my_s0_server --type m5d.2xlarge"
+    echo "   $me --k /path/to/key.pem -n my_s0_server -t m5d.2xlarge"
+    echo
+    echo "  -k | --key-name : A name of an aws EC2 key in the PEM format for the SSH connection to the instance."
+    echo
+    echo "  -n | --name     : Optional. Name of an instance. It's shown in the AWS console."
+    echo "                    The default value is \"[user_name]-s0-server\"."
+    echo
+    echo "  -t | --type     : Optional. An instance type. See https://aws.amazon.com/ec2/instance-types/ for more details."
+    echo "                    The default value is \"m5d.2xlarge\"."
+    echo
+    echo "  -h | --help     : Show this usage help."
+}
+
+[ "$1" == "" ] && echo "Script need some arguments!" && usage && exit -1
+
+while [ "$1" != "" ]; do
+    case $1 in
+        -k | --key-name)   shift && key_name=$1 ;;
+        -n | --name)       shift && instance_name=$1 ;;
+        -t | --type)       shift && instance_type=$1 ;;
+        -h | --help)       usage && exit ;;
+        *)                 echo "Wrong arguments!"
+                           usage && exit 15 ;;
+    esac
+        shift
+done
+
+set -e
+
+if [ -z "$key_name" ]; then
+    echo "Please specify the '--key-name' parameter. Otherwise, you won't be able to connect to the running instance via ssh!"
+    echo
+    usage
+    exit 1
+fi
+
+[ -z "$instance_name" ] && instance_name=$default_instance_name
+[ -z "$instance_type" ] && instance_type=$default_instance_type
+
+latest_ami=$(aws ec2 describe-images --owners $elastio_aws_id \
+    --filters "Name=name,Values=s0-ami*" "Name=state,Values=available" \
+    --query "reverse(sort_by(Images, &CreationDate))[:1].ImageId" --output text)
+
+if aws ec2 describe-security-groups | jq '.SecurityGroups[].GroupName' | tr -d '"' | grep -q "$security_group" ; then
+    sg_id=$(aws ec2 describe-security-groups --group-names $security_group --query "SecurityGroups[:1].GroupId" --output text)
+else
+    if ! sg_id=$(aws ec2 create-security-group --group-name $security_group --description "Ports 22 and 61234 are open to the world for s0 server and ssh" --output text) ; then
+        echo "Failed to create a security group to open ports for s0 server and ssh. Do you have enough permissions?"
+        exit 2
+    fi
+    if ! aws ec2 authorize-security-group-ingress --group-id $sg_id --protocol tcp --port 22 --cidr 0.0.0.0/0 ||
+       ! aws ec2 authorize-security-group-ingress --group-id $sg_id --protocol tcp --port 61234 --cidr 0.0.0.0/0 ; then
+        echo "Failed to open ports 22 and 61234 for s0 server and ssh!"
+        exit 3
+    fi
+fi
+
+echo "Launching \"$instance_name\" (instance type $instance_type) with the latest Elastio s0 AMI ($latest_ami)"
+
+instance_json=$(aws ec2 run-instances \
```

Comment:
The `launch-instance.sh` script in `elastio/elastio` also uses cloud-init to format the locally attached NVMe drive and install some packages.  Obviously the package install stuff doesn't make sense here since this is our AMI, but setting up the NVMe disk and mounting it somewhere would also be a valuable addition.

On the call today we also talked about scripting the rest of the process, which is launching `s0` itself and serving a vault given the bucket name, KMS key alias, and shard type (fixed or variable).    Do you plan to do that in a separate script?  It would be a better user experience if they're together.

Reply:
Ok, what size that disk should be? And why we need a disk to add? Does s0 use an s3 bucket and doesn't use a local storage?

I've started to work on this script in terms of https://github.com/elastio/packaging/issues/40 before today's decision to make it extended with the `s0` configuration. And there is a new issue created after the discussion by @mvovchuk https://github.com/elastio/elastio/issues/1326 which have to be discussed. For now I have no idea how s0 should be configured.
My preference to get a script which just starts an instance as of https://github.com/elastio/packaging/issues/40 and then extend it to config s0 as of https://github.com/elastio/elastio/issues/1326. I plan to have all of it in this script, in two steps.

Reply:
done, see the list here https://github.com/elastio/elastio/issues/1326